src/PlaneWaveBasis.jl

```diff
@@ -12,8 +12,8 @@ More generally, a kpoint is a block of the Hamiltonian;
 eg collinear spin is treated by doubling the number of kpoints.
 """
 struct Kpoint{T <: Real}
-    model::Model{T}
-    spin::Symbol                     # :up, :down, :both or :spinless
+    model::Model{T}                  # TODO Should be only lattice/atoms
```

Comment:
I'm perfectly fine with it being the full model here...

Reply:
I'm not. This should just be the structure.